mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -210,7 +218,7 @@ private void cleanup(JobContext jobContext) throws IOException {
     JobConf conf = jobContext.getJobConf();
     // If there are reducers, then every reducer will generate a result file.
     // If this is a map only task, then every mapper will generate a result file.
-    int expectedFiles = conf.getNumReduceTasks() != 0 ? conf.getNumReduceTasks() : conf.getNumMapTasks();
+    int expectedFiles = conf.getNumReduceTasks() > 0 ? conf.getNumReduceTasks() : conf.getNumMapTasks();
```

Comment:
Is this fixing a bug?

Reply:
yes, exactly. For map-only tasks, `numReduceTasks` can have -1 value, which would lead to `expectedFiles` being -1 as well